CMakeLists.txt

```diff
@@ -36,6 +37,16 @@ if(WIN32)
   target_compile_definitions(${PROJECT_NAME}
     PRIVATE "RCPPUTILS_BUILDING_LIBRARY")
 endif()
+
+option(USE_PI_MUTEX "Enables priority inheritance mutexes." ON)
+if(USE_PI_MUTEX)
+  if(WIN32 OR APPLE)
```

Comment:
I would do it other way around. I would list systems with PI_MUTEX support and fail if the system is not in the list. With this approach you will get an error for unknown systems and for new systems you explicitly add them when you know it is supported.

Reply:
Makes sense. Better now?

Unfortunately I could not find much info on the CMAKE_SYSTEM_NAME variable for VxWorks and QNX systems. I hope I got the right strings.

Reply:
@razr FYI